activerecord/test/cases/associations/has_and_belongs_to_many_associations_test.rb

```diff
@@ -577,6 +577,11 @@ def test_find_in_association_with_options
     assert_equal developers(:poor_jamis), projects(:active_record).developers.where("salary < 10000").first
   end
 
+  test "test_association_with_extend_option" do
+    post = posts(:welcome)
```

Comment:
@senny  It is difficult to define an anonymous subclass. `Builder::HasAndBelongsToMany#middle_reflection` requires class name.
So, I just define a class on the test file.